### ISSUE TITLE
QUIP can't read file written by ASE with a info entry that's set to None

### ISSUE THREAD

**Issue by noambernstein** [libAtoms/QUIP#22]
I'm not sure if this is best described as a QUIP or ASE I/O issue, but if you use the ASE extxyz writer to create a file for an atoms object that has an info entry which has value None, QUIP (e.g. convert.py) can't read it.  The info entry becomes a bare word in the comment line (KEmax below), which is apparently not parseable.

tin 1027 : cat t.py 

``` python
#!/usr/bin/env python

import ase, ase.io

a = ase.Atoms('CO', positions=[(0, 0, 0), (0, 0, 1)])

a.info['KEmax'] = None

ase.io.write("tt.xyz", a, format="extxyz")
```

tin 1028 : python t.py 
tin 1029 : cat tt.xyz

```
2
Lattice="1.0 0.0 0.0 0.0 1.0 0.0 0.0 0.0 1.0" Properties=species:S:1:pos:R:3:Z:I:1 KEmax pbc="F F F"
C      0.00000000       0.00000000       0.00000000        6 
O      0.00000000       0.00000000       1.00000000        8 
```

tin 1030 : convert.py tt.xyz -

``````
Usage: convert.py [options] ( <input file> [ <output file> ] | <input file> [ <input file> ... ] [ (-o|--output) <output file> ] )

convert.py: error: Traceback (most recent call last)
  File "/usr/local/quip/src/QUIP/src/libAtoms/xyz.c", line 732 kind IO
    Badly formed key/value pair KEmax                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         ```
``````

**Comment by jameskermode** [comment]
I think this is a QUIP issue. In `args_str` entries, keys with missing values imply a value of `.true.`. Would it be reasonable to apply the same default here?

**Comment by noambernstein** [comment]
Well, on _write_ it's not exactly a missing value, it's "None".  Reading it
as True wouldn't work for the case I'm using, no, unless I change the code
to test for something other than "at.info['KEmax'] is None", or if I custom
check the input info dict and convert the True to None.

**Comment by jameskermode** [comment]
Keys without a value should get type T_NONE in libAtoms/Dictionary.f95, and this should map to None in Python.

**Comment by noambernstein** [comment]
Ok - then the issue is that it's not written/read consistently, at least not if ase is involved. Should I check all the read and write and ase and quip combinations?

**Comment by albapa** [comment]
I am wondering if anyone is working on this issue at the moment - if not, I'd be keen, because I'd like to run NS-MC with quip and I'm told this is currently not possible.

**Comment by jameskermode** [comment]
I'm having a look now. I think it's a fairly simple fix to the parser in xyz.c to treat bare keys as type T_NONE.

**Comment by jameskermode** [comment]
Should be fixed in commit 10245cffc56cf5 - please have a look at let me know if there are any further problems.

**Comment by jameskermode** [comment]
Closing as no further problems reported.
